### ISSUE TITLE
Optional arguments don't work

### ISSUE THREAD

**Issue by skissane** [IBM/JSONata4Java#176]
See this [JSONata expression (1.8.5)](https://try.jsonata.org/old-Bxbg1):
```
 ( $x := function($a,$b) { [$a, $b ? $b : "default" ] }; $x(32) )
```

Produces output:
```
[
  32,
  "default"
]
```

But in JSONata4Java, using `tester.sh` on latest master branch (commit efd22732f9588231f4ae7284e7d1b400f9fa1494), you instead get the error:
```
Expected equal counts for varibles (2) and values (1)
```

(A related, but very minor, issue, is that the error message contains a spelling mistake, `varibles` instead of `variables`.)

**Comment by wnm3** [comment]
I've patched the code to allow only throw the exception if the passed variables exceed the function declaration, using null for the missing values. This seemed to pass your example and didn't break the test cases from jsonata.org.

You can clone then build using the `mvn clean install` command in the JSONata4Java directory to get the jar files built in the target directory.

Please test and close if you believe this solves the problem and I'll add this to the next release of JSONata4Java.

**Comment by skissane** [comment]
I tested it, that works, thanks. And thank you for the fast response!

A couple of minor comments (not that important):
- when I pass too many arguments, I still get the `varibles` misspelling in the error message
- `mvn clean install` doesn't work for me, I get an error about `gpg` not being installed. I could install `gpg`, but I don't really care about signing it, so I just did `mvn clean install -Dgpg.skip` instead
